toolkit/tools/pkg/imagecustomizerlib/artifactsinputoutput.go

```diff
@@ -220,36 +211,23 @@ func InjectFiles(buildDir string, baseConfigPath string, inputImageFile string,
 ) error {
 	logger.Log.Debugf("Injecting Files")
 
-	ic := &ImageCustomizerParameters{}
 	buildDirAbs, err := filepath.Abs(buildDir)
 	if err != nil {
 		return err
 	}
-	ic.inputImageFile = inputImageFile
-	ic.inputImageFormat = strings.TrimLeft(filepath.Ext(ic.inputImageFile), ".")
-	ic.rawImageFile = filepath.Join(buildDirAbs, BaseImageName)
-	ic.outputImageFile = ic.inputImageFile
-	ic.outputImageFormat = imagecustomizerapi.ImageFormatType(ic.inputImageFormat)
-	if err := ic.outputImageFormat.IsValid(); err != nil {
+	inputImageFormat := strings.TrimLeft(filepath.Ext(inputImageFile), ".")
+	rawImageFile := filepath.Join(buildDirAbs, BaseImageName)
+	outputImageFormat := imagecustomizerapi.ImageFormatType(inputImageFormat)
```

Comment:
We should detect the file format by reading the input file instead of looking at the file extension. Look at the `convertImageToRaw` and `getImageFileInfo` functions.

Reply:
Fixed. Please have a review.